ADR/0023-release-and-releaseplan-authorization.md

```diff
@@ -0,0 +1,55 @@
+# Release and ReleasePlan Authorization
+
+Date: 2023-04-17
+
+## Status
+
+Accepted
+
+## Context
+
+* It is imperative to know which associate has authorized a change to production in order to have accountability for changes.
+* Flexibility is required when specifying authorization either on a per-release basis or as a standing authorization.
+
+## Decision
+
+### Architecture Overview
+#### Terminology
+
+- `Release`: A CRD that describes what should be released and using which plan.
+- `ReleasePlan`: A CRD that describes where and how an application should be released.
+- `ReleaseStrategy`: A CRD that describes which pipeline to execute along with specific pipeline parameters.
+
+#### Architecture
+
+![](../diagrams/ADR-0023/ReleaseAndReleasePlanAuthorization.jpg)
+
+##### Capturing authorization data
+* The idea is that when a release pipeline executes, the `Pipeline`'s `Tasks` have access to the name of the associate that authorized the change in production.
+* `Authorization` for a particular `Release` should be obtained by capturing the user that has created the object.
+  * The `status` section of the `Release` CR should be updated with to specify the `author`.
+* A `Standing Authorization` for an application should be obtained by capturing the user that creates or updates the `ReleasePlan` CR **when** `annotations.standing-authorization` is set to `true`
+  * Since we expect ReleasePlans to be updated by users who may not intend to provide a `Standing Authorization` (i.e. Change `ReleaseStrategy`), it is important to intercept only when the above annotation is set.
+  * The `status` section of the `ReleasePlan` CR should be updated with the `author`.
+
+##### Processing a Release
+* When a `Release` is processed, the Release service must follow this process:
+  * Verify that the `status.author` is populated.
```

Comment:
Is it possible to prevent this from being set by the user themselves? If not, this could be tampered with to claim sign-off by a different author.

Reply:
The status section of the CR can only be changed by the controller